sentry-ruby/lib/sentry/scope.rb

```diff
@@ -74,6 +89,7 @@ def dup
       copy.tags = tags.deep_dup
       copy.user = user.deep_dup
       copy.transaction_names = transaction_names.deep_dup
+      copy.transaction_sources = transaction_sources.deep_dup
```

Comment:
I think this and the `transaction_names` don't need `deep_dup`, just `dup` should be enough. Especially in the `transaction_sources`'s case, the content are all `Symbol` and are not mutable.



I can't recall why I used `deep_dup` at `transaction_names`, but given that strings are generally frozen nowadays and there's no point mutating them in practice, I think it can switch to `dup` too.